test/gurka/test_turn_lanes.cc

```diff
@@ -393,6 +393,145 @@ TEST(Standalone, TurnLanesUTurns) {
+      {{
+           {gurka::way_member, "CF", "from"},
+           {gurka::way_member, "GB", "to"},
+           {gurka::way_member, "FG", "via"},
+       },
+       {
+           {"type", "restriction"},
+           {"restriction", "no_u_turn"},
+       }},
+      {{
+           {gurka::way_member, "FL", "from"},
+           {gurka::way_member, "IG", "to"},
+           {gurka::way_member, "LI", "via"},
+       },
+       {
+           {"type", "restriction"},
+           {"restriction", "no_u_turn"},
+       }},
+  };
+
+  const auto layout =
+      gurka::detail::map_to_coordinates(ascii_map, gridsize_metres, {5.1079374, 52.0887174});
+
+  auto map = gurka::buildtiles(layout, ways, {}, relations, "test/data/gurka_turn_lanes_7");
+
+  valhalla::Api result;
+
+  // Test U-turn using left-only lanes. The left-most left lane should be
+  // active & rest invalid
+  result = gurka::route(map, "A", "H", "auto");
+
+  gurka::assert::raw::expect_maneuvers(result, {DirectionsLeg_Maneuver_Type_kStart,
+                                                DirectionsLeg_Maneuver_Type_kUturnLeft,
+                                                DirectionsLeg_Maneuver_Type_kDestination});
+  validate_turn_lanes(result, {
+                                  {5, "[ *left* ACTIVE | left | through | right | right ]"},
+                                  {5, "[ left | left | *through* ACTIVE | right | right ]"},
+                                  {0, ""},
+                                  {0, ""},
+                                  {0, ""},
+                              });
+
+  // Test using reverse lanes. The reverse lane should be active & the rest
+  // invalid
+  result = gurka::route(map, "E", "D", "auto");
+  gurka::assert::raw::expect_maneuvers(result, {DirectionsLeg_Maneuver_Type_kStart,
+                                                DirectionsLeg_Maneuver_Type_kUturnLeft,
+                                                DirectionsLeg_Maneuver_Type_kDestination});
+  validate_turn_lanes(result, {
+                                  {4, "[ *reverse* ACTIVE | left | through | right ]"},
+                                  {4, "[ *reverse* ACTIVE | left | through | right ]"},
+                                  {0, ""},
+                                  {5, "[ left | left | *through* ACTIVE | right | right ]"},
+                                  {0, ""},
+                              });
+
+  result = gurka::route(map, "K", "J", "auto");
+
+  gurka::assert::raw::expect_maneuvers(result, {DirectionsLeg_Maneuver_Type_kStart,
+                                                DirectionsLeg_Maneuver_Type_kUturnLeft,
+                                                DirectionsLeg_Maneuver_Type_kDestination});
+  validate_turn_lanes(result,
+                      {
+                          {0, ""},
+                          {0, ""},
+                          {3, "[ *through* ACTIVE | *through* VALID | right ]"}, // restriction does
```

Comment:
Maybe I'm missing something obvious, but I don't see any lane info on the `KC` segment, so where did these lanes come from?

Reply:
These lanes are coming from LM.   This test adds restrictions to the map so that I can confirm that lanes are not being inferred when there is a complex restriction.  So basically, the restriction forces the route to go straight on 
KCFLM and make a left onto MO and continue straight to J via OIGBJ